cub/block/radix_rank_sort_operations.cuh

```diff
@@ -130,16 +119,518 @@ struct ShiftDigitExtractor : BaseDigitExtractor<KeyT>
 {
     using typename BaseDigitExtractor<KeyT>::UnsignedBits;
 
-    uint32_t bit_start, mask;
+    std::uint32_t bit_start;
+    std::uint32_t mask;
+
     explicit __device__ __forceinline__ ShiftDigitExtractor(
-        uint32_t bit_start = 0, uint32_t num_bits = 0)
-        : bit_start(bit_start), mask((1 << num_bits) - 1)
+        std::uint32_t bit_start = 0, std::uint32_t num_bits = 0)
+      : bit_start(bit_start)
+      , mask((1 << num_bits) - 1)
     { }
 
-    __device__ __forceinline__ uint32_t Digit(UnsignedBits key)
+    __device__ __forceinline__ std::uint32_t Digit(UnsignedBits key) const
+    {
+        return std::uint32_t(this->ProcessFloatMinusZero(key) >> UnsignedBits(bit_start)) & mask;
+    }
+};
+
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS    // Do not document
+namespace detail 
+{
+
+template <bool... Bs>
+struct logic_helper_t;
+
+template <bool>
+struct true_t 
+{
+  static constexpr bool value = true;
+};
+
+template <bool... Bs>
+using all_t =            //
+  ::cuda::std::is_same<  //
+    logic_helper_t<Bs...>, //
+    logic_helper_t<true_t<Bs>::value...>>;
+
+struct fundamental_decomposer_t
```

Comment:
I am wondering whether `identity_decomposer_t` would be better, but I am really not sure whether that is a better name

Reply:
I think `identity_decomposer_t` is better. I'll rename it, thanks!